services/venice-controller/src/main/java/com/linkedin/venice/controller/StoreDeletionValidationUtils.java

```diff
@@ -0,0 +1,331 @@
+      if (storeConfig != null) {
+        result.setStoreNotDeleted("Store config still exists in storeConfigRepo.");
+        return true;
+      }
+    } catch (Exception e) {
+      LOGGER.warn("Failed to check store config for store: {} in cluster: {}", storeName, clusterName, e);
+      result.setStoreNotDeleted("Failed to check store config: " + e.getMessage());
+      return true;
+    }
+    return false;
+  }
+
+  /**
+   * Checks if store metadata still exists in the store repository.
+   */
+  private static boolean checkStoreMetadata(
+      Admin admin,
+      StoreDeletedValidation result,
+      String clusterName,
+      String storeName) {
+    try {
+      final Store store = admin.getStore(clusterName, storeName);
+      if (store != null) {
+        result.setStoreNotDeleted("Store metadata still exists in storeRepository.");
+        return true;
+      }
+    } catch (Exception e) {
+      LOGGER.warn("Failed to check store metadata for store: {} in cluster: {}", storeName, clusterName, e);
+      result.setStoreNotDeleted("Failed to check store metadata: " + e.getMessage());
+      return true;
+    }
+    return false;
+  }
+
+  /**
+   * Checks if system store metadata still exists.
+   */
+  private static boolean checkSystemStoreMetadata(
+      Admin admin,
+      StoreDeletedValidation result,
+      String clusterName,
+      String storeName) {
+    try {
+      final List<VeniceSystemStoreType> systemStoreTypesToCheck = Arrays.asList(VeniceSystemStoreType.values())
+          .stream()
+          .filter(type -> type != VeniceSystemStoreType.BATCH_JOB_HEARTBEAT_STORE) // Not per-store
+          .collect(Collectors.toList());
+
+      for (VeniceSystemStoreType systemStoreType: systemStoreTypesToCheck) {
+        final String systemStoreName = systemStoreType.getSystemStoreName(storeName);
+        final Store systemStore = admin.getStore(clusterName, systemStoreName);
+        if (systemStore != null) {
+          result.setStoreNotDeleted(
+              String.format("System store metadata still exists in storeRepository: %s", systemStoreName));
+          return true;
+        }
+      }
+    } catch (Exception e) {
+      LOGGER.warn("Failed to check system store metadata for store: {} in cluster: {}", storeName, clusterName, e);
+      result.setStoreNotDeleted("Failed to check system store metadata: " + e.getMessage());
+      return true;
+    }
+    return false;
+  }
+
+  /**
+   * Checks if Kafka topics related to the store still exist.
+   */
+  private static boolean checkForAnyExistingTopicResources(
+      Admin admin,
+      StoreDeletedValidation result,
+      String clusterName,
+      String storeName) {
+    try {
+      final TopicManager topicManager = admin.getTopicManager();
+      final Set<PubSubTopic> kafkaTopics = topicManager.listTopics();
+
+      final List<VeniceSystemStoreType> systemStoreTypesToCheck = Arrays.asList(VeniceSystemStoreType.values())
```

Comment:
if this call is frequent, let's replace it with looping constructs. Streams generate to much garbage